### DISCUSSION TITLE
Switch only one heater on - not two

### DISCUSSION THREAD

**Discussion by jlockham** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#123] [Ideas]
Firstly let me thank you for this amazing ‘hack’/add-on!

I note that the pump units have two heaters in them - and by observing power usage - when the heater is first turned on - it only turns one heater on for a bit, and then turns the second heater on after a period of time.

I'm not sure if the ESP with this code on can control the independent heaters or not - but this would be really cool if we could via mqtt - so I can use solar power (at different stages generation watt's) to heat the spa (say first watt step would be 1200w, then the second heater stage would be when generating 2200w or more. I can do the stages in HA with node red – but just need the ability to turn the heaters on independently.

If the ESP cant do this (due to the main PCB controlling this) then I can add in a second relay unit to the wiring – could we add to the code for the ESP to control two addition Digital pins to control this relay (like you have already for the solar input and the external temp sensors)?

Thanks in advance!

**Comment by visualapproach** [comment]
@jlockham You are welcome! 
Hardware-wise you can use the two spare GPIOs on the PCB to do whatever you want. The Solar input example is just an example. Beware though - they are 3.3v and cannot deliver more than a few mA. You need some sort of amplification to drive a relay. Possibly enough with a transistor and a base resistor, but you need to get power to the relay from somewhere. Not sure i would trust the 5V line to cope with the extra load.
Software-wise it is doable. Either by adding another MQTT command, or sensing an input.
I agree with you that it would be a nice feature, but as the pump is designed it requires some surgery on 230V mains components. I will not include that in this project for safety reasons, but it should be relatively straight forward if you know what you are doing.

**Comment by visualapproach** [comment]
Actually, I found that on some 4-wire models we can turn only one, or both heaters. The 6-wire models is not possible without hardware changes.

**Comment by Snotmann** [comment]
Hey together ...

i got the same question ... there seems to be two heater units and my solar system could provide 1kw power. Can i put only one heater on so that i dont have to use energy from the grid to heat the pool ?

@visualapproach 

Or is it possible by disamble some wires from one of the heater modules ?

Regards,
Daniel

**Reply by visualapproach** [reply]
Depends on what model you have. 4-wire can use the command full power 0.
Else disconnect one heater.

**Reply by Snotmann** [reply]
@visualapproach ... I got the 6 wire model ... Ok behind the cover where the 4 white cables come from sinply disconnect. I will try rhat

**Reply by visualapproach** [reply]
Yup. Disconnect everything from mains first to stay alive. And tape the cables or something to prevent shorts

**Reply by Snotmann** [reply]
@visualapproach i did that ... Isolated the cables to ... Takes 1000 Watt now ... Thats for the Tipp ... Saves a Lot of Money and CO2 👍😅
